### PR TITLE
feat(notifier): Issue Body Parser Module (Story 2.2.1)

### PR DESCRIPTION
**Pull request by nam20485**
## Summary

Implements Story 2.2.1 of Epic #38 - Intelligent Template Triaging.

### Changes

- **Extended TaskType enum** in `src/models/work_item.py`:
  - Added `BUG`, `FEATURE`, `ENHANCEMENT`, `GENERIC` types
  - Provides richer classification beyond PLAN/IMPLEMENT

- **Created Issue Body Parser** in `src/notifier/parsers/issue_parser.py`:
  - Regex-based template detection for common headers:
    - `[Application Plan]`, `[Bug]`, `[Feature]`, `[Enhancement]`
  - Content-based detection with confidence scoring
  - Label and title fallback detection
  - Support for custom patterns

- **Added comprehensive unit tests** in `tests/unit/test_issue_parser.py`:
  - 49 test cases covering all template types
  - Edge cases (empty body, malformed markdown)
  - Real-world issue examples

### Test Results

```
tests/unit/test_issue_parser.py ... 49 passed
All 377 tests pass
```

## Acceptance Criteria

- [x] Create `src/notifier/parsers/issue_parser.py`
- [x] Implement regex patterns for common template headers
- [x] Map headers to TaskType enum values
- [x] Add unit tests with sample issue bodies
- [x] Handle malformed markdown gracefully

Part of #38

### CODE REVIEW

File: src/notifier/parsers/issue_parser.py [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+        result = parser.parse("# [Application Plan]\\nThis is a plan...")
+        print(result.detected_type)  # TaskType.PLAN
+        ```
+    """
+
+    # Regex patterns for template detection
+    # Patterns are ordered by specificity - more specific patterns first
+    TEMPLATE_PATTERNS: list[tuple[str, TaskType]] = [
+        # Application Plan patterns
+        (r"^\s*#\s*\[?Application Plan\]?", TaskType.PLAN),
```
[the remark this answers is not in the document]
Comment (nam20485) [reply src/notifier/parsers/issue_parser.py]: ✅ Fixed in aee9888

Issue: Regex pattern was too permissive, matching malformed headers with unbalanced brackets.
Resolution: Updated TEMPLATE_PATTERNS to use conditional regex that ensures brackets are properly paired. Applied to all patterns handling optional brackets (Application Plan, App Plan, Plan, Bug, Bug Report, Feature, Feature Request, Enhancement).
Notes: Verified with Python tests - patterns now correctly reject unbalanced brackets like '# [Bug' while accepting both '# [Bug]' and '# Bug'.

File: src/notifier/parsers/issue_parser.py [hunk 2]
```diff
@@ -0,0 +1,411 @@
...
+        Args:
+            labels: List of label names.
+
+        Returns:
+            ParseResult if a label indicates a type, None otherwise.
+        """
+        label_lower = [label.lower() for label in labels]
+
+        # Check for type-indicating labels
+        if any("bug" in label for label in label_lower):
```
[the remark this answers is not in the document]
Comment (nam20485) [reply src/notifier/parsers/issue_parser.py]: ✅ Fixed in aee9888

Issue: Label check using substring matching was too broad, causing false positives like 'debug' or 'ladybug' matching 'bug'.
Resolution: Changed to exact match using 'label == "bug"' pattern. Applied same logic to 'feature', 'enhancement', and 'plan'/'epic' label checks.

File: src/notifier/parsers/issue_parser.py [hunk 3]
```diff
@@ -0,0 +1,411 @@
...
+
+    detected_type: TaskType
+    confidence: float = 1.0
+    matched_pattern: str | None = None
+    raw_header: str | None = None
+
+    model_config = {
+        "frozen": False,
+        "extra": "forbid",
+    }
```
[the remark this answers is not in the document]
Comment (nam20485) [reply src/notifier/parsers/issue_parser.py]: ✅ Fixed in aee9888

Issue: The model_config dict in ParseResult dataclass was inspired by Pydantic but has no effect on standard Python dataclass - misleading/dead code.
Resolution: Removed the model_config dictionary entirely from the ParseResult class.

File: src/notifier/parsers/issue_parser.py [hunk 4]
```diff
@@ -0,0 +1,411 @@
...
+            title: Optional issue title.
+
+        Returns:
+            ParseResult containing the detected type and metadata.
+        """
+        # First try body parsing
+        result = self.parse(body)
+
+        # If high confidence match, return it
+        if result.confidence >= 0.8:
```
[the remark this answers is not in the document]
Comment (nam20485) [reply src/notifier/parsers/issue_parser.py]: ✅ Fixed in aee9888

Issue: Confidence threshold 0.8 was hardcoded in parse_with_fallback method.
Resolution: Defined HIGH_CONFIDENCE_THRESHOLD = 0.8 as a class-level constant and used it in the comparison.

File: src/notifier/parsers/issue_parser.py [hunk 5]
```diff
@@ -0,0 +1,411 @@
...
+
+    Example:
+        >>> result = parse_issue_body("# [Bug] Something is broken")
+        >>> result.detected_type
+        <TaskType.BUG: 'BUG'>
+    """
+    global _default_parser
+    if _default_parser is None:
+        _default_parser = IssueBodyParser()
+    return _default_parser.parse(body)
```
[the remark this answers is not in the document]
Comment (nam20485) [reply src/notifier/parsers/issue_parser.py:390]: ✅ Fixed in aee9888

Issue: Module-level _default_parser initialization was not thread-safe - race condition could cause multiple IssueBodyParser instantiations.
Resolution: Added threading.Lock (_parser_lock) with double-checked locking pattern in both parse_issue_body() and parse_issue_with_context() functions. Added 'import threading' at module level.

**Comment by nam20485** [comment]
## PR Review Comments Addressed ✅

All review comments from @gemini-code-assist have been addressed in commit aee9888.

### Summary of Changes

| # | Priority | Issue | Resolution |
|---|----------|-------|------------|
| 1 | HIGH | Regex patterns too permissive (unbalanced brackets) | Used conditional regex `(\[)?...(?(1)\])` to ensure balanced brackets |
| 2 | HIGH | Label check too broad (false positives) | Changed to exact match: `label == "bug"` |
| 3 | MEDIUM | Dead `model_config` dict | Removed entirely from ParseResult dataclass |
| 4 | MEDIUM | Hardcoded confidence threshold | Defined `HIGH_CONFIDENCE_THRESHOLD = 0.8` as class constant |
| 5 | MEDIUM | Thread-unsafe parser initialization | Added `threading.Lock` with double-checked locking |

### Thread Status

- ✅ Thread safety comment (line 386) - **Resolved**
- ℹ️ Other comments are marked as **outdated** by GitHub (lines changed) - replies posted but not resolved per protocol

### CI Status

All checks passing:
- ✅ lint
- ✅ scan  
- ✅ test
- ✅ CodeQL

Ready for re-review and merge approval.
